bazel/pocket-ic-tests.bzl

```diff
@@ -92,3 +92,35 @@ pocket_ic_mainnet_test = rule(
     executable = True,
     test = True,
 )
+
+def pocket_ic_test(name, test_rule, **kwargs):
```

Comment:
I'm not too fond of small macros that just wrap a target and inject dependencies; I find they tend to add complexity and obfuscate what is really happening, making everything harder to follow and debug, especially when e.g. the `test_rule` needs to be specified.

Is the goal here mostly to avoid repetition? Do we have a ticket we can link to re. the sandbox issue on Darwin?

Reply:
Context is in this [thread](https://dfinity.slack.com/archives/CL7Q2RXUM/p1734601302312279). I would like to have a central place where we do work-arounds for that sandbox issue on darwin and not scatter it all over our code base.

Reply:
I see; agree it's a good idea to have a central place for the workaround but I don't think this requires implicitly injecting `pocket-ic`. How about modifying `rust_ic_test`  to add `requires-network` if `pocket-ic` is a dependency?

Also, can you add a ticket so that we can track the issue upstream and remove the workaround once it's fixed?

Reply:
> How about modifying rust_ic_test to add requires-network if pocket-ic is a dependency?

That seems rather hacky and an implicit rule like that could lead to surprises. Also it's not just `rust_ic_test` that needs to be modified but also `rust_test`, `rust_test_suite` and `rust_test_suite_with_extra_srcs`. Or do they all resolve back to `rust_test`?

> can you add a ticket so that we can track the issue upstream and remove the workaround once it's fixed?

Good idea: https://dfinity.atlassian.net/browse/IDX-3192

Reply:
Sorry, I meant a ticket upstream on bazel! With just an internal ticket I doubt we'll follow up & make progress on this

Reply:
This seems like a similar issue: https://github.com/bazelbuild/bazel/issues/14828. Let me study that a bit more later.

Reply:
There seems to be [a work around](https://github.com/bazelbuild/bazel/issues/5206#issuecomment-402398624) by using IPv6 instead of IPv4. @mraszyk maybe that's something you can look into.